File: file-templates/templates/MVVM/UITableViewCellModel.xctemplate/UITableViewCellXIBObjective-C/___FILEBASENAME___.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

@interface ___FILEBASENAMEASIDENTIFIER___ ()
@property (nonatomic, weak) IBOutlet UIView *containerView;
@end

@implementation ___FILEBASENAMEASIDENTIFIER___

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

#pragma mark - private

- (void)onUpdateUIComponents {
    
}

- (void)onSetupBinding {
    [self onUpdateUIComponents];
}

#pragma mark - setter

- (void)setCellModel:(___VARIABLE_productName___Model *)cellModel {
    if (_cellModel != cellModel) {
        _cellModel = cellModel;
        [self onSetupBinding];
    }
}

@end

File: file-templates/templates/MVVM/Cell.xctemplate/UITableViewCellXIBSwift/___FILEBASENAME___.swift
//___FILEHEADER___

import UIKit

class ___FILEBASENAMEASIDENTIFIER___: UITableViewCell {
    
    // MARK: - Porperties

    public var provider: ___FILEBASENAMEASIDENTIFIER___ModelProvider? {
        didSet {
            setupBindings()
        }
    }
    
    /// 容器视图
    @IBOutlet var containerView: UIView!

    // MARK: - override
    
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
        onInitUIComponents()
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

    
    // MARK: - UIComponents
    
    public func onInitUIComponents() {

    }

    public func setupBindings() {
        onUpdateUIComponents()
    }
    
    public func onUpdateUIComponents() {
        guard let provider = self.provider else { return }
        // Configure components for provider
    }
}
